src/nano_contracts/deserializer.ts

```diff
@@ -170,6 +185,20 @@ class Deserializer {
     signedBuffer = signedBuffer.slice(size);
     return `${bufferToHex(signedBuffer)},${parsed},${valueType}`;
   }
+
+  /**
+   * Deserialize a value decoded in bytes to a base58 string
+   *
+   * @param {value} Value to deserialize
+   *
+   * @memberof Deserializer
+   * @inner
+   */
+  toAddress(value: Buffer): string {
+    // First we get the 20 bytes of the address without the version byte and checksum
+    const addressBytes = value.slice(1, 21);
+    return helpersUtils.encodeAddress(addressBytes, this.network).base58;
```

Comment:
question: Since we're discarding the first byte of the input here, are we at risk of accepting an address from another network as valid?

Would it be worth it to add a test validating the behavior of having an address from another network being decoded?

Reply:
Makes sense! 

Done
https://github.com/HathorNetwork/hathor-wallet-lib/pull/729/commits/12200527419855290ae7ede7ebd0e35555c41039
https://github.com/HathorNetwork/hathor-wallet-lib/pull/729/commits/cd74c357d0f39b0c153651cc8394a27f7661c888